bundles/org.openhab.ui/web/src/components/widgets/layout/oh-grid-layout.vue

```diff
@@ -162,8 +159,8 @@ export default {
           for (x = 0; x < this.colNum; x++) {
             free = true
             for (let i = 0; i < this.layout.length; i++) {
-              if (!((this.layout[i].x + this.layout[i].w <= x) || (this.layout[i].x >= x + 1) ||
-                    (this.layout[i].y + this.layout[i].h <= y) || (this.layout[i].y >= y + 1))) {
+              if (!((this.layout[i].x + this.layout[i].w <= x) || (this.layout[i].x >= x + 2) ||
+                    (this.layout[i].y + this.layout[i].h <= y) || (this.layout[i].y >= y + 2))) {
```

Comment:
Hi @ghys, 
Unfortunately, it's not that simple to get 2x2 widgets: This adds a 2-width element to a column when there's only a 1-width spot left, same for rows. As a result widgets overlap or exceed the defined screen.

Reply:
Heh- shouldn't be too hard to fix. The 1x1 defaults widgets were too small, I thought.
Also the documentation should be generated (and ideally manually completed) for the new components, and the layout page documentation page should mention the new features.

Reply:
> Heh- shouldn't be too hard to fix. The 1x1 defaults widgets were too small, I thought.

Surely possible, though one issue always remains: What if there only remains a 1x1 spot or 1x3 row somewhere in the middle of the layout and the user wants to fill that (e.g. on a 3x5 (button) grid created for a mobile phone)? The logic would then have to detect that there's only space for 1x1 widgets left and add such instead.

I aggree that a 1x1 widget can be quite small. Another way to fix this would be to simply default to less columns, e.g. 8 or 12 instead of currently 16.

For now I'd ask you to revert your change, as it results in this:
![add_widgets_issue](https://user-images.githubusercontent.com/2242688/110213323-d72ecb00-7e9f-11eb-8812-4c69c7c0c87a.gif)

Reply:
> Surely possible, though one issue always remains: What if there only remains a 1x1 spot or 1x3 row somewhere in the middle of the layout and the user wants to fill that (e.g. on a 3x5 (button) grid created for a mobile phone)? The logic would then have to detect that there's only space for 1x1 widgets left and add such instead.

Exactly, this doesn't seem that complex to implement.

> I aggree that a 1x1 widget can be quite small. Another way to fix this would be to simply default to less columns, e.g. 8 or 12 instead of currently 16.

Preferably not since there might be cases where the a widget thinner (or smaller) than the default could be desired.